contracts/build.rs

```diff
@@ -0,0 +1,41 @@
+use ethcontract_generate::Builder;
+use std::{env, fs, path::Path};
+
+#[path = "src/paths.rs"]
+mod paths;
+
+fn main() {
+    // NOTE: This is a workaround for `rerun-if-changed` directives for
+    // non-existant files cause the crate's build unit to get flagged for a
+    // rebuild if any files in the workspace change.
+    //
+    // See:
+    // - https://github.com/rust-lang/cargo/issues/6003
+    // - https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath
+    println!("cargo:rerun-if-changed=build.rs");
+
+    generate_contract("IERC20");
+}
+
+fn generate_contract(name: &str) {
+    let artifact = paths::contract_artifacts_dir().join(format!("{}.json", name));
+    let address_file = paths::contract_address_file(name);
+    let dest = env::var("OUT_DIR").unwrap();
+
+    println!("cargo:rerun-if-changed={}", artifact.display());
+    let mut builder = Builder::new(artifact)
+        .with_visibility_modifier(Some("pub"))
+        .add_event_derive("serde::Deserialize")
+        .add_event_derive("serde::Serialize");
+
+    if let Ok(address) = fs::read_to_string(&address_file) {
+        println!("cargo:rerun-if-changed={}", address_file.display());
+        builder = builder.add_deployment_str(5777, address.trim());
+    }
+
```

Comment:
this might not be needed since we don't deploy in Rust. But it might be worth leaving it in if the eventually happens.

Reply:
True. I will remove it until we need it at which point we can copy it back from dex-services.